### PR TITLE
3318 add SmithAndKoothoor2016 reference for SRS template

### LINKED ISSUE #3318: Add SmithAndKoothoor2016 reference for SRS template
# TODO
Smith and Koothoor (2016) (which can be found in [References.bib](https://gitlab.cas.mcmaster.ca/smiths/pub/-/blob/master/References.bib) in the pub repo) should be referenced alongside Smith and Lai (2005) and Smith et al. (2007) for the SRS template in the Organization of Document section of all applicable examples.

## _Initial question:_
While working on ChemCode, I realized that the SRS template provided by @smiths found [here](https://github.com/smiths/capTemplate/blob/main/docs/SRS/SRS.pdf) is based on Smith and Lai (2005) and Smith et al. (2007), where as our Drasil SRSs are based on Smith and Lai (2005) and Koothoor (2013). Should Smith et al. (2007) be added to the rest of our Drasil examples (and should Koothoor (2013) be added to the capTemplate SRS)?

### PR DESCRIPTION
**Pull request by harmanpreet-sagar**
Closes #3318: Add SmithAndKoothoor2016 reference for SRS template. Changes have been made to stable, example, and the Citations code.

### CODE REVIEW

File: code/drasil-example/dblpendulum/lib/Drasil/DblPendulum/Body.hs [hunk 1]
```diff
@@ -189,8 +189,8 @@ organizationOfDocumentsIntro :: Sentence
 organizationOfDocumentsIntro = foldlSent 
   [atStartNP (the organization), S "of this", phrase document, 
   S "follows the", phrase template, S "for an", getAcc Doc.srs, S "for", 
-  phrase sciCompS, S "proposed by", refS koothoor2013 `S.and_`
-  refS smithLai2005]
+  phrase sciCompS, S "proposed by", refS koothoor2013 `sC` refS smithLai2005 
+  `sC` refS smithEtAl2007 `S.and_` refS smithKoothoor2016]
```
Comment (samm82) [inline code/drasil-example/dblpendulum/lib/Drasil/DblPendulum/Body.hs:193]: Could you make a separate <kbd>[newcomers](https://github.com/JacquesCarette/Drasil/labels/newcomers)</kbd> issue for investigating whether this sentence could be reused between examples?

Comment (samm82) [inline code/drasil-example/dblpendulum/lib/Drasil/DblPendulum/Body.hs:193]: Also, this list should use the `foldlList` function; I believe something like
```suggestion
  phrase sciCompS, S "proposed by", foldlList Comma List $
    map refS [koothoor2013, smithLai2005, smithEtAl2007, smithKoothoor2016]
```

File: code/drasil-example/gamephysics/lib/Drasil/GamePhysics/Body.hs [hunk 2]
```diff
@@ -39,7 +39,7 @@ import Drasil.GamePhysics.Concepts (gamePhysics, acronyms, threeD, twoD)
 import Drasil.GamePhysics.DataDefs (dataDefs)
 import Drasil.GamePhysics.Goals (goals)
 import Drasil.GamePhysics.IMods (iMods, instModIntro)
-import Drasil.GamePhysics.References (citations, koothoor2013, smithLai2005)
+import Drasil.GamePhysics.References (citations, koothoor2013, smithEtAl2007, smithLai2005, smithKoothoor2016)
```
Comment (samm82) [inline code/drasil-example/gamephysics/lib/Drasil/GamePhysics/Body.hs:42]: Try to keep line lengths below 80 characters where possible:
```suggestion
import Drasil.GamePhysics.References (citations, koothoor2013, smithEtAl2007,
  smithLai2005, smithKoothoor2016)
```

File: code/stable/dblpendulum/SRS/JSON/DblPendulum_SRS.ipynb [hunk 3]
```diff
@@ -167,7 +167,7 @@
     "## Organization of Document\n",
     "<a id=\"Sec:DocOrg\"></a>\n",
     "\n",
-    "The organization of this document follows the template for an SRS for scientific computing software proposed by <a href=#koothoor2013>koothoor2013</a>  and <a href=#smithLai2005>smithLai2005</a> . The presentation follows the standard pattern of presenting goals, theories, definitions, and assumptions. For readers that would like a more bottom up approach, they can start reading the [instance models](#Sec:IMs) and trace back to find any additional information they require.\n",
+    "The organization of this document follows the template for an SRS for scientific computing software proposed by <a href=#koothoor2013>koothoor2013</a> , <a href=#smithLai2005>smithLai2005</a> , <a href=#smithEtAl2007>smithEtAl2007</a>  and <a href=#smithKoothoor2016>smithKoothoor2016</a> . The presentation follows the standard pattern of presenting goals, theories, definitions, and assumptions. For readers that would like a more bottom up approach, they can start reading the [instance models](#Sec:IMs) and trace back to find any additional information they require.\n",
```
Comment (samm82) [inline code/stable/dblpendulum/SRS/JSON/DblPendulum_SRS.ipynb:170]: It looks like there are extra spaces being added after the references for Jupyter notebooks; if this isn't already a <kbd>[newcomers](https://github.com/JacquesCarette/Drasil/labels/newcomers)</kbd> issue, it should be!

File: code/drasil-data/lib/Data/Drasil/Citations.hs [hunk 4]
```diff
@@ -52,6 +52,21 @@ smithLai2005 = cInProceedings [spencerSmith, lLai]
...
+smithEtAl2007 = cArticle [spencerSmith, lLai, rKhedri]
+  ("Requirements Analysis for Engineering Computation: A Systematic Approach for" ++ 
+    " Improving Software Reliability")
+  "Reliable Computing, Special Issue on Reliable Engineering Computation" 2007
+  [month Feb, volume 13, number 1, pages [83..107], howPublishedU "https://doi.org/10.1007/s11155-006-9020-7"]
+  "smithEtAl2007"
+
+smithKoothoor2016 = cArticle [spencerSmith, nKoothoor]
+  ("A Document-Driven Method for Certifying Scientific Computing Software for Use" ++ 
+    "in Nuclear Safety Analysis")
```
Comment (samm82) [inline code/drasil-data/lib/Data/Drasil/Citations.hs:64]: You're missing a space here, which is also affecting stable:
```suggestion
    " in Nuclear Safety Analysis")
```

**Review by samm82 — CHANGES_REQUESTED** [review]

File: code/drasil-example/dblpendulum/lib/Drasil/DblPendulum/Body.hs (hunk 1 above)
Comment (harmanpreet-sagar) [reply code/drasil-example/dblpendulum/lib/Drasil/DblPendulum/Body.hs:193]: Yeah for sure! The issue that I have created for investigating this is #3410.

**Review by smiths — COMMENTED** [review]
The changes to stable look good to me.

File: code/drasil-example/glassbr/lib/Drasil/GlassBR/References.hs [hunk 5]
```diff
@@ -2,13 +2,14 @@ module Drasil.GlassBR.References where
...
+import Data.Drasil.Citations (campidelli, koothoor2013, smithEtAl2007, smithLai2005, 
+  smithKoothoor2016, parnasClements1986)
 import Data.Drasil.People (jRobertson, jmBracci, sRobertson, tlKohutek, wlBeason)
 
 rbrtsn2012, astm2009, astm2016, astm2012, beasonEtAl1998 :: Citation
 
 citations :: BibRef
-citations = [campidelli, koothoor2013, smithLai2005, rbrtsn2012, astm2009, astm2016,
+citations = [campidelli, koothoor2013, smithEtAl2007, smithLai2005, smithKoothoor2016, rbrtsn2012, astm2009, astm2016,
   astm2012, beasonEtAl1998, parnasClements1986]
```
Comment (samm82) [inline code/drasil-example/glassbr/lib/Drasil/GlassBR/References.hs:13]: Watch your line lengths!
```suggestion
citations = [campidelli, koothoor2013, smithEtAl2007, smithLai2005,
  smithKoothoor2016, rbrtsn2012, astm2009, astm2016, astm2012, beasonEtAl1998,
  parnasClements1986]
```

**Review by samm82 — CHANGES_REQUESTED** [review]
There are still some lines throughout this PR that are too long (a minor nitpick, I know) but other than that this looks great!

File: code/drasil-example/gamephysics/lib/Drasil/GamePhysics/Body.hs (hunk 2 above)
Comment (smiths) [reply code/drasil-example/gamephysics/lib/Drasil/GamePhysics/Body.hs]: @harmanpreet-sagar, to help with line length, you can install Rewrap in VSCode (https://stackoverflow.com/questions/43122175/automatically-hard-wrap-lines-at-column-in-vscode).  I believe this is one of the steps in setting up a new workspace, but you might not have set the line length.  The usual suggestion is 80 characters.  Use Alt+Q to rewrap a paragraph with hard returns.

File: code/drasil-example/glassbr/lib/Drasil/GlassBR/References.hs (hunk 5 above)
Comment (harmanpreet-sagar) [reply code/drasil-example/glassbr/lib/Drasil/GlassBR/References.hs:13]: I will check those out right now!

**Review by samm82 — APPROVED** [review]
There are still some outlier cases of long line lengths, but the majority of them are in files that should have their formatting improved anyways or aren't a result of changes from this PR, so I'm OK to approve this PR (unless @smiths thinks we should go all the way with line lengths!)

**Pull request merged**